packages/api/src/routes/ehr/shared.ts

```diff
@@ -20,21 +17,16 @@ export type ParseResponse = {
 
 export async function processCxIdAsync(
   req: Request,
-  source: EhrSources.athena | EhrSources.canvas,
+  tokenSource: EhrDashJwtTokenSource | EhrWebhookJwtTokenSource,
   parseExternalId: (tokenData: JwtTokenData) => ParseResponse
 ): Promise<void> {
   const accessToken = getAuthorizationToken(req);
-  const authInfo = await getJwtToken({
-    token: accessToken,
-    source,
-  });
+  const authInfo = await getJwtToken({ token: accessToken, source: tokenSource });
   if (!authInfo) throw new ForbiddenError();
   if (authInfo.exp < new Date()) throw new ForbiddenError();
   const { externalId, queryParams } = parseExternalId(authInfo.data);
-  const customer = await getCxMappingOrFail({
-    externalId,
-    source,
-  });
+  const cxMappingSource = getCxMappingSourceFromJwtTokenSource(tokenSource);
+  const customer = await getCxMappingOrFail({ externalId, source: cxMappingSource });
```

Comment:
These functions might throw 5xx or 4xx, including information about why it failed. But this function is an auth function IIUIC, so we should have a global try/catch and always return the same forbidden error - [reference](https://www.notion.so/metriport/Development-Best-Practices-bb3574d3c83d46dc9914a9e5c5542ca7?pvs=4#1adb51bb90408097a437e5e44c0d5565)